src/corehost/cli/deps_resolver.cpp

```diff
@@ -289,71 +274,52 @@ bool deps_resolver_t::probe_entry_in_configs(const deps_entry_t& entry, pal::str
         {
             if (entry.to_hash_matched_path(probe_dir, candidate))
             {
-                assert(!config.is_roll_fwd_set());
                 trace::verbose(_X("    Matched hash for [%s]"), candidate->c_str());
                 return true;
             }
             trace::verbose(_X("    Skipping... match hash failed"));
         }
         else if (config.probe_deps_json)
         {
-            // If the deps json has it then someone has already done rid selection and put the right stuff in the dir.
-            // So checking just package name and version would suffice. No need to check further for the exact asset relative path.
+            // If the deps json has the package name and version, then someone has already done rid selection and
+            // put the right asset in the dir. So checking just package name and version would suffice.
+            // No need to check further for the exact asset relative sub path.
             if (config.probe_deps_json->has_package(entry.library_name, entry.library_version) && entry.to_dir_path(probe_dir, candidate))
             {
-                trace::verbose(_X("    Probed deps json and matched [%s]"), candidate->c_str());
+                trace::verbose(_X("    Probed deps json and matched '%s'"), candidate->c_str());
                 return true;
             }
             trace::verbose(_X("    Skipping... probe in deps json failed"));
         }
-        else if (!config.is_roll_fwd_set())
+        else if (config.probe_publish_dir)
```

Comment:
Because, we don't roll forward for individual packages anywhere. So it was all dead code.